Fix py-tests CI

**Affected module**
python ingestion

**Describe the bug**
```
=========================== short test summary info ============================
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_chart_api.py::OMetaChartTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_dashboard_api.py::OMetaDashboardTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_database_service_api.py::OMetaDatabaseServiceTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_delete
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_entity_ref
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_entity_version
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_id
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_get_name
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_list_versions
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_mlmodel_properties
FAILED ingestion/tests/integration/ometa/test_ometa_model_api.py::OMetaModelTest::test_update
================== 24 failed, 70 passed, 2 warnings in 6.80s ===================
```

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**
A clear and concise description of what you expected to happen.

**Version:**
 - OS: [e.g. iOS]
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**
Add any other context about the problem here.

--- a/ingestion/src/metadata/ingestion/ometa/mixins/version_mixin.py
+++ b/ingestion/src/metadata/ingestion/ometa/mixins/version_mixin.py
@@ -13,7 +13,7 @@ from requests.models import Response
 from metadata.generated.schema.type import basic
 from metadata.generated.schema.type.entityHistory import EntityVersionHistory
 from metadata.ingestion.ometa.client import REST
-from metadata.ingestion.ometa.utils import uuid_to_str
+from metadata.ingestion.ometa.utils import model_str
 
 T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
 logger = logging.getLogger(__name__)
@@ -68,7 +68,7 @@ class OMetaVersionMixin(Generic[T]):
         fields: List
             List of fields to return
         """
-        entity_id = uuid_to_str(entity_id)
+        entity_id = model_str(entity_id)
         version = self.version_to_str(version)
 
         path = f"{entity_id}/versions/{version}"
@@ -95,7 +95,7 @@ class OMetaVersionMixin(Generic[T]):
         List
             lists of available versions for a specific entity
         """
-        path = f"{uuid_to_str(entity_id)}/versions"
+        path = f"{model_str(entity_id)}/versions"
 
         resp = self.client.get(f"{self.get_suffix(entity)}/{path}")
 

--- a/ingestion/src/metadata/ingestion/ometa/ometa_api.py
+++ b/ingestion/src/metadata/ingestion/ometa/ometa_api.py
@@ -64,7 +64,7 @@ from metadata.ingestion.ometa.openmetadata_rest import (
     NoOpAuthenticationProvider,
     OktaAuthenticationProvider,
 )
-from metadata.ingestion.ometa.utils import fqdn_to_str, get_entity_type, uuid_to_str
+from metadata.ingestion.ometa.utils import model_str, get_entity_type
 
 logger = logging.getLogger(__name__)
 
@@ -396,7 +396,7 @@ class OpenMetadata(
         Return entity by name or None
         """
 
-        return self._get(entity=entity, path=f"name/{fqdn_to_str(fqdn)}", fields=fields)
+        return self._get(entity=entity, path=f"name/{model_str(fqdn)}", fields=fields)
 
     def get_by_id(
         self,
@@ -408,7 +408,7 @@ class OpenMetadata(
         Return entity by ID or None
         """
 
-        return self._get(entity=entity, path=uuid_to_str(entity_id), fields=fields)
+        return self._get(entity=entity, path=model_str(entity_id), fields=fields)
 
     def _get(
         self, entity: Type[T], path: str, fields: Optional[List[str]] = None
@@ -463,7 +463,7 @@ class OpenMetadata(
             return EntityReference(
                 id=instance.id,
                 type=get_entity_type(entity),
-                name=fqdn_to_str(instance.fullyQualifiedName),
+                name=model_str(instance.fullyQualifiedName),
                 description=instance.description,
                 href=instance.href,
             )
@@ -509,7 +509,7 @@ class OpenMetadata(
         """
 
         suffix = self.get_suffix(entity)
-        path = f"/{uuid_to_str(entity_id)}/versions"
+        path = f"/{model_str(entity_id)}/versions"
         resp = self.client.get(f"{suffix}{path}")
 
         if self._use_raw_data:
@@ -542,7 +542,7 @@ class OpenMetadata(
         Returns
             None
         """
-        url = f"{self.get_suffix(entity)}/{uuid_to_str(entity_id)}"
+        url = f"{self.get_suffix(entity)}/{model_str(entity_id)}"
         url += f"?recursive=true" if recursive else ""
         self.client.delete(url)
 

--- a/ingestion/src/metadata/ingestion/ometa/ometa_api.py
+++ b/ingestion/src/metadata/ingestion/ometa/ometa_api.py
@@ -64,7 +64,7 @@ from metadata.ingestion.ometa.openmetadata_rest import (
     NoOpAuthenticationProvider,
     OktaAuthenticationProvider,
 )
-from metadata.ingestion.ometa.utils import get_entity_type, uuid_to_str, fqdn_to_str
+from metadata.ingestion.ometa.utils import fqdn_to_str, get_entity_type, uuid_to_str
 
 logger = logging.getLogger(__name__)
 
@@ -387,7 +387,10 @@ class OpenMetadata(
         return entity_class(**resp)
 
     def get_by_name(
-        self, entity: Type[T], fqdn: Union[str, FullyQualifiedEntityName], fields: Optional[List[str]] = None
+        self,
+        entity: Type[T],
+        fqdn: Union[str, FullyQualifiedEntityName],
+        fields: Optional[List[str]] = None,
     ) -> Optional[T]:
         """
         Return entity by name or None

--- a/ingestion/src/metadata/ingestion/ometa/ometa_api.py
+++ b/ingestion/src/metadata/ingestion/ometa/ometa_api.py
@@ -45,6 +45,7 @@ from metadata.generated.schema.entity.teams.role import Role
 from metadata.generated.schema.entity.teams.team import Team
 from metadata.generated.schema.entity.teams.user import User
 from metadata.generated.schema.type import basic
+from metadata.generated.schema.type.basic import FullyQualifiedEntityName
 from metadata.generated.schema.type.entityHistory import EntityVersionHistory
 from metadata.generated.schema.type.entityReference import EntityReference
 from metadata.ingestion.ometa.auth_provider import AuthenticationProvider
@@ -63,7 +64,7 @@ from metadata.ingestion.ometa.openmetadata_rest import (
     NoOpAuthenticationProvider,
     OktaAuthenticationProvider,
 )
-from metadata.ingestion.ometa.utils import get_entity_type, uuid_to_str
+from metadata.ingestion.ometa.utils import get_entity_type, uuid_to_str, fqdn_to_str
 
 logger = logging.getLogger(__name__)
 
@@ -386,13 +387,13 @@ class OpenMetadata(
         return entity_class(**resp)
 
     def get_by_name(
-        self, entity: Type[T], fqdn: str, fields: Optional[List[str]] = None
+        self, entity: Type[T], fqdn: Union[str, FullyQualifiedEntityName], fields: Optional[List[str]] = None
     ) -> Optional[T]:
         """
         Return entity by name or None
         """
 
-        return self._get(entity=entity, path=f"name/{fqdn}", fields=fields)
+        return self._get(entity=entity, path=f"name/{fqdn_to_str(fqdn)}", fields=fields)
 
     def get_by_id(
         self,
@@ -459,7 +460,7 @@ class OpenMetadata(
             return EntityReference(
                 id=instance.id,
                 type=get_entity_type(entity),
-                name=instance.fullyQualifiedName,
+                name=fqdn_to_str(instance.fullyQualifiedName),
                 description=instance.description,
                 href=instance.href,
             )

--- a/ingestion/src/metadata/ingestion/ometa/ometa_api.py
+++ b/ingestion/src/metadata/ingestion/ometa/ometa_api.py
@@ -64,7 +64,7 @@ from metadata.ingestion.ometa.openmetadata_rest import (
     NoOpAuthenticationProvider,
     OktaAuthenticationProvider,
 )
-from metadata.ingestion.ometa.utils import model_str, get_entity_type
+from metadata.ingestion.ometa.utils import get_entity_type, model_str
 
 logger = logging.getLogger(__name__)
 

--- a/ingestion/src/metadata/ingestion/ometa/utils.py
+++ b/ingestion/src/metadata/ingestion/ometa/utils.py
@@ -19,6 +19,7 @@ from typing import Type, TypeVar, Union
 from pydantic import BaseModel
 
 from metadata.generated.schema.type import basic
+from metadata.generated.schema.type.basic import FullyQualifiedEntityName
 
 T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
 
@@ -66,3 +67,17 @@ def uuid_to_str(entity_id: Union[str, basic.Uuid]) -> str:
         return str(entity_id.__root__)
 
     return entity_id
+
+
+def fqdn_to_str(fqdn: Union[str, FullyQualifiedEntityName]) -> str:
+    """
+    Given a FQDN, that can be a str or our pydantic
+    definitions of FullyQualifiedEntityName. Return a
+    str to build the endpoint path.
+    :param fqdn: FullyQualifiedEntityName or str
+    :return: string of FullyQualifiedEntityName
+    """
+    if isinstance(fqdn, basic.FullyQualifiedEntityName):
+        return str(fqdn.__root__)
+
+    return fqdn

--- a/ingestion/src/metadata/ingestion/ometa/utils.py
+++ b/ingestion/src/metadata/ingestion/ometa/utils.py
@@ -14,6 +14,7 @@ Helper functions to handle OpenMetadata Entities' properties
 
 import re
 import string
+from functools import singledispatch
 from typing import Type, TypeVar, Union
 
 from pydantic import BaseModel
@@ -55,29 +56,19 @@ def get_entity_type(
     return class_name
 
 
-def uuid_to_str(entity_id: Union[str, basic.Uuid]) -> str:
+@singledispatch
+def model_str(arg) -> str:
     """
-    Given an entity_id, that can be a str or our pydantic
-    definition of Uuid, return a proper str to build
-    the endpoint path
-    :param entity_id: Entity ID to onvert to string
-    :return: str for the ID
+    Default model stringifying method
     """
-    if isinstance(entity_id, basic.Uuid):
-        return str(entity_id.__root__)
-
-    return entity_id
+    return str(arg)
 
 
-def fqdn_to_str(fqdn: Union[str, FullyQualifiedEntityName]) -> str:
+@model_str.register(basic.Uuid)
+@model_str.register(basic.FullyQualifiedEntityName)
+@model_str.register(basic.EntityName)
+def _(arg) -> str:
     """
-    Given a FQDN, that can be a str or our pydantic
-    definitions of FullyQualifiedEntityName. Return a
-    str to build the endpoint path.
-    :param fqdn: FullyQualifiedEntityName or str
-    :return: string of FullyQualifiedEntityName
+    Models with __root__
     """
-    if isinstance(fqdn, basic.FullyQualifiedEntityName):
-        return str(fqdn.__root__)
-
-    return fqdn
+    return str(arg.__root__)

--- a/ingestion/src/metadata/ingestion/ometa/utils.py
+++ b/ingestion/src/metadata/ingestion/ometa/utils.py
@@ -20,7 +20,6 @@ from typing import Type, TypeVar, Union
 from pydantic import BaseModel
 
 from metadata.generated.schema.type import basic
-from metadata.generated.schema.type.basic import FullyQualifiedEntityName
 
 T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
 

--- a/ingestion/tests/unit/test_ometa_utils.py
+++ b/ingestion/tests/unit/test_ometa_utils.py
@@ -16,7 +16,7 @@ from unittest import TestCase
 
 from metadata.generated.schema.entity.data.mlmodel import MlModel
 from metadata.generated.schema.type import basic
-from metadata.ingestion.ometa.utils import format_name, get_entity_type, uuid_to_str
+from metadata.ingestion.ometa.utils import format_name, get_entity_type, model_str
 
 
 class OMetaUtilsTest(TestCase):
@@ -37,13 +37,16 @@ class OMetaUtilsTest(TestCase):
         self.assertEqual(get_entity_type("hello"), "hello")
         self.assertEqual(get_entity_type(MlModel), "mlmodel")
 
-    def test_uuid_to_str(self):
+    def test_model_str(self):
         """
         Return Uuid as str
         """
 
-        self.assertEqual(uuid_to_str("random"), "random")
+        self.assertEqual(model_str("random"), "random")
         self.assertEqual(
-            uuid_to_str(basic.Uuid(__root__="9fc58e81-7412-4023-a298-59f2494aab9d")),
+            model_str(basic.Uuid(__root__="9fc58e81-7412-4023-a298-59f2494aab9d")),
             "9fc58e81-7412-4023-a298-59f2494aab9d",
         )
+
+        self.assertEqual(model_str(basic.EntityName(__root__="EntityName")), "EntityName")
+        self.assertEqual(model_str(basic.FullyQualifiedEntityName(__root__="FQDN")), "FQDN")

--- a/ingestion/tests/unit/test_ometa_utils.py
+++ b/ingestion/tests/unit/test_ometa_utils.py
@@ -48,5 +48,9 @@ class OMetaUtilsTest(TestCase):
             "9fc58e81-7412-4023-a298-59f2494aab9d",
         )
 
-        self.assertEqual(model_str(basic.EntityName(__root__="EntityName")), "EntityName")
-        self.assertEqual(model_str(basic.FullyQualifiedEntityName(__root__="FQDN")), "FQDN")
+        self.assertEqual(
+            model_str(basic.EntityName(__root__="EntityName")), "EntityName"
+        )
+        self.assertEqual(
+            model_str(basic.FullyQualifiedEntityName(__root__="FQDN")), "FQDN"
+        )

